app/main.py

```diff
@@ -1,3 +1,45 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if type(other) in (int, float):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other):
```

Comment:
But usually `iadd` is `+=`. So it will not change your code working, but better replace `self.km = self.km + other` with `self.km += other` and so on